src/color-man.cc

```diff
@@ -699,6 +699,7 @@ guchar *heif_color_profile(FileData *fd, guint *profile_len)
 	if (error_code.code)
 		{
 		log_printf("warning: heif reader error: %s\n", error_code.message);
+		heif_image_handle_release(handle);
```

Comment:
Consider using [smart pointers](https://docs.gtk.org/glib/auto-cleanup.html) to avoid duplication of this block. Example: https://github.com/BestImageViewer/geeqie/blob/db6271995d8d9e408329f47a50d198706ad3cf9d/src/image-load-j2k.cc#L39C1-L41C62

Reply:
On second thought libheif has C++ API. It's probably a good time to use it.